plaso/parsers/google_logging.py

```diff
@@ -0,0 +1,264 @@
+
+    Args:
+      parser_mediator (ParserMediator): mediates interactions between parsers
+          and other components, such as storage and dfvfs.
+      structure (pyparsing.ParseResults): elements parsed from the file.
+    """
+    month = self._GetValueFromStructure(structure, 'month_number')
+
+    if month != 0:
+      self._UpdateYear(parser_mediator, month)
+
+    day = self._GetValueFromStructure(structure, 'day')
+    hours = self._GetValueFromStructure(structure, 'hour')
+    minutes = self._GetValueFromStructure(structure, 'minute')
+    seconds = self._GetValueFromStructure(structure, 'second')
+    microseconds = self._GetValueFromStructure(structure, 'microsecond')
+
+    time_elements_tuple = (
+        self._year_use, month, day, hours, minutes, seconds, microseconds)
+
+    try:
+      date_time = dfdatetime_time_elements.TimeElementsInMicroseconds(
+          time_elements_tuple=time_elements_tuple)
+      date_time.is_local_time = True
+    except ValueError:
+      parser_mediator.ProduceExtractionWarning(
+          'invalid date time value: {0!s}'.format(time_elements_tuple))
+      return
+
+    event_data = GoogleLogEventData()
+    event_data.priority = self._GetValueFromStructure(structure, 'priority')
+    event_data.thread_id = self._GetValueFromStructure(structure, 'thread_id')
+    event_data.file_name = self._GetValueFromStructure(structure, 'file_name')
+    event_data.line_number = self._GetValueFromStructure(
+        structure, 'line_number')
+    event_data.message = self._GetValueFromStructure(structure, 'message')
+
+    event = time_events.DateTimeValuesEvent(
+        date_time, definitions.TIME_DESCRIPTION_WRITTEN)
+    parser_mediator.ProduceEventWithEventData(event, event_data)
+
+  def ParseRecord(self, parser_mediator, key, structure):
+    """Parses a matching entry.
+
+    Args:
+      parser_mediator (ParserMediator): mediates interactions between parsers
+          and other components, such as storage and dfvfs.
+      key (str): name of the parsed structure.
+      structure (pyparsing.ParseResults): elements parsed from the file.
+
+    Raises:
+      ParseError: when the structure type is unknown.
+    """
+    if key not in self._SUPPORTED_KEYS:
+      raise errors.ParseError(
+          'Unable to parse record, unknown structure: {0:s}'.format(key))
+
+    if key == 'greeting':
+      self._ReadGreeting(structure)
+
+    elif key == 'log_entry':
+      self._ParseLine(parser_mediator, structure)
+
+  def VerifyStructure(self, parser_mediator, lines):
+    """Verifies that this is a google log-formatted file.
+
+    Args:
+      parser_mediator (ParserMediator): mediates interactions between
+          parsers and other components, such as storage and dfvfs.
+      lines (str): one or more lines from the text file.
+
+    Returns:
+      bool: True if this is the correct parser, False otherwise.
+    """
+    if not lines.startswith(self._GREETING_START):
+      return False
+
```

Comment:
If the "greeting" contains a year and month value let's check if these are valid as well.

Reply:
maybe validate the entire date?

Reply:
Is there a particular false positive case you're concerned about? Adding the validation is going to complicate the logic quite a bit, and I think the likelihood and impact of starting to parse a file with a different format are very low.

Reply:
So `Log file created at: ` IMHO is not that unique. I can imagine that other log formats use this as well. Hence the request to add additional validation.

Reply:
Could you do something with the additional header lines:
```
Running on machine: plasotest1
Binary: Built on May 14 2019 17:08:01 (1368576481)
Binary: Built for gcc-4.X.Y
Log line format: [IWEF]mmdd hh:mm:ss.uuuuuu threadid file:line] msg
```

Reply:
Did a rough search but most of the tools seem to use glog under the hood. Also
```
Binary: Built on May 14 2019 17:08:01 (1368576481)
Binary: Built for gcc-4.X.Y"
```
Seem to be not required lines